### ISSUE TITLE
`deno repl` does not support typescript

### ISSUE THREAD

**Issue by realharry** [denoland/deno#3809]
Deno claims to be a "secure runtime for JavaScript and TYPESCRIPT", and it is one of the main reasons why I started looking into Deno.

And yet, a simple test on `deno repl` shows that the typescript is not really supported. Is this by design? Is this Deno's technical limitations?

For me, personally, being able to transpile Typescript to Javascript has relatively low value. Whether I compile the code via tsc or deno (with embedded tsc), or any other tools, has very little difference. It's just a build process. 

The real value for me (if Deno supports Typescript as a first class citizen, as it seems to promise) will be being able to "hide" the transpilation process from the user experience, like in REPL.

**Comment by realharry** [comment]
I did a very simple test to check if typescript was supported in REPL:

```
> var a:string = 'Hello Deno!';
error: Uncaught SyntaxError: Unexpected token ':'
► <unknown>:1:6
    at evaluate ($deno$/repl.ts:83:34)
    at replLoop ($deno$/repl.ts:174:13)
```

**Comment by ry** [comment]
It's a technical challenge to overcome. It will feel laggy if we compile TS in the REPL. @bartlomieju actually has a patch for it here https://github.com/denoland/deno/pull/3760, but I'm torn because most of the time in a REPL you don't care about type and rather want one-off calculations.

Deno tries to be "JavaScript first" - meaning that there is much less overhead to running JS and JS users shouldn't be hindered. In the case of the REPL, adding a TS compile would hinder someone just wanting to do "1 + 2".

Ultimately we need to figure out how to speed up TS compilation.

Duplicate of https://github.com/denoland/deno/issues/1158, so closing.

**Comment by J-Cake** [comment]
Perhaps a typescript REPL could be enabled with a CLI flag? If you start the REPL with `--ts` flag or execute a REPL command, you'll get typescript.

**Comment by ertw** [comment]
This is unexpected behavior for me too. My naive expectation was something similar to the Elm repl, which (and I'm not sure why) always seems slow to load initially, but is then fast to evaluate expressions. If you haven't tried the Elm repl, you can give it a spin in the browser here: http://elmrepl.cuberoot.in/

```elm
> two = 1 + 1                                                                                                                                                                                                     
2 : number                                                                                                                                                                                                        
>                                                                                                                                                                                                                 
```

In this example, the Elm compiler is able to understand the types, and gives me an answer with the type signature. This would be my "perfect world" dream for the Deno repl. If doing a full compile pass is too slow, would it be possible to at least run queries through the type stripper so that I could copy and paste TS into the repl?

**Comment by kitsonk** [comment]
@ertw this issue is closed as a duplicate of #1158.
